xc/xc7/yosys/utils.tcl

```diff
@@ -0,0 +1,22 @@
+# Update the CLKOUT[0-5]_PHASE parameter values.
+# Due to the fact that Yosys doesn't support floating parameter values
+# i.e. treats them as strings, the parameter values need to be multiplied by 1000
+# for the PLL registers to have correct values calculated during techmapping.
+proc update_pll_params {} {
+    set temp_file "[pid].tmp"
+    select t:PLLE2_ADV -list -write $temp_file
+    set fh [open $temp_file r]
+    foreach cell [gets $fh] {
+	for {set output 0} {$output < 6} {incr output} {
+	    set param_name "CLKOUT${output}_PHASE"
+	    set param_value [getparam $param_name t:PLLE2_ADV]
+	    if {$param_value > 0} {
```

Comment:
I think that this condition should be removed. I'm not sure about that but the value of "0.0" may not be interpreter by Yosys as integer 0. So its better to process it always.

Reply:
The concern here was more about the param value being empty.
I changed the condition to `ne ""` (non-empty string) so the `0.0` get converted to 0.